test/Microsoft.Health.Fhir.Tests.E2E/Rest/CompartmentTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Tests.Common;
+using Microsoft.Health.Fhir.Tests.E2E.Common;
+using Microsoft.Health.Fhir.Web;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Rest.Search
+{
+    public class CompartmentTests : SearchTestsBase<HttpIntegrationTestFixture<Startup>>, IAsyncLifetime
+    {
+        private Patient _patient;
+        private Observation _observation;
+        private Device _device;
+        private DeviceComponent _deviceComponent;
+        private Encounter _encounter;
+        private Condition _condition;
+        private readonly string _tagValue;
+        private const string TagSearchParameterName = "_tag";
+        private string _tagQueryString;
+
+        public CompartmentTests(HttpIntegrationTestFixture<Startup> fixture)
+            : base(fixture)
+        {
+            _tagValue = $"CompartmentTests_{Guid.NewGuid().ToString()}";
+            _tagQueryString = $"{TagSearchParameterName}={_tagValue}";
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenADeviceCompartment_WhenRetrievingDeviceComponents_ThenOnlyResourcesMatchingCompartmentShouldBeReturned()
+        {
+            Bundle bundle = await Client.SearchAsync("Device/d1/DeviceComponent");
```

Comment:
I don't think these test works. If you are creating a new resource now, you will get new server generated id but these tests still uses hard-coded client id?

Reply:
These tests still work because while creating reference search indices we do not check the existence of the reference resource. I will change to using PUT instead of POST to avoid the confusion.